src/models/inbound/dfspOTPValidate.model.ts

```diff
@@ -0,0 +1,356 @@
+          // first unsubscribe
+          this.pubSub.unsubscribe(channel, sid)
+
+          this.data.consentsStatus = { ...message as unknown as tpAPI.Schemas.ErrorInformation }
+          resolve()
+        })
+
+        this.logger.push({ res }).info('ThirdpartyRequests.postConsents request sent to peer')
+
+        resolve()
+      } catch (error) {
+      }
+    })
+    */
+
+  }
+
+  /**
+   * depending on current state of model returns proper result
+   */
+  getResponse (): void {
+    switch (this.data.currentState) {
+      default:
+        return
+    }
+  }
+
+  reformatError (err: Error): Errors.MojaloopApiErrorObject {
+    let mojaloopErrorCode = Errors.MojaloopApiErrorCodes.INTERNAL_SERVER_ERROR
+    if (err instanceof HTTPResponseError) {
+      const e = err.getData()
+      if (e.res && (e.res.body || e.res.data)) {
+        if (e.res.body) {
+          try {
+            const bodyObj = JSON.parse(e.res.body)
+            mojaloopErrorCode = Errors.MojaloopApiErrorCodeFromCode(`${bodyObj.statusCode}`)
+          } catch (ex) {
+            // do nothing
+            this.logger.push({ ex }).error('Error parsing error message body as JSON')
+          }
+        } else if (e.res.data) {
+          mojaloopErrorCode = Errors.MojaloopApiErrorCodeFromCode(`${e.res.data.statusCode}`)
+        }
+      }
+    }
+
+    return new Errors.MojaloopFSPIOPError(
+      err,
+      null as unknown as string,
+      null as unknown as string,
+      mojaloopErrorCode
+    ).toApiErrorObject()
+  }
+
+  /**
+   * runs the workflow
+   */
+  async run (): Promise<void> {
+    const data = this.data
+    try {
+      // run transitions based on incoming state
+      switch (data.currentState) {
+        case 'start':
+          await this.saveToKVS()
+          this.logger.info(
+            `validateOTP requested for ${data.consentRequestsRequestId},  currentState: ${data.currentState}`
+          )
+          await this.fsm.validateOTP()
+          await this.saveToKVS()
+          break
+
+        case 'OTPIsValid':
+          this.logger.info(
+            `requestScopes requested for ${data.consentRequestsRequestId},  currentState: ${data.currentState}`
+          )
+          await this.fsm.requestScopes()
+          await this.saveToKVS()
+          break;
+
+        case 'scopesReceived':
+          this.logger.info(
+            `sendConsent requested for ${data.consentRequestsRequestId},  currentState: ${data.currentState}`
+          )
+          await this.fsm.sendConsent()
+          await this.saveToKVS()
+          // workflow is finished
+          return
+
+        default:
+          this.logger.info('State machine in errored state')
+          return
+      }
+      this.logger.info(`Transfer model state machine transition completed in state: ${this.fsm.state}. Recursing to handle next transition.`)
+      this.run()
```

Comment:
We are making here a recursive call that could be dangerous in javascript.
Recursion must be guarded. 
I can see that only two `cases` end with `break`, and the rest end with `return`. 
So there is a guard and its logic is spread across the switch cases - fine, but code layout could be improved:

I suggest replacing `break` with `return this.run()` so the whole case logic will be together in one block and there will be no need to remember that there is a recursion call after the switch statement or revising the switch logic when reader see the recursion call after the switch statement.

Reply:
done.

Reply:
hmm did you pushed changes? looks I am still seeing old version...

Reply:
yes I updated `break` to `return this.run()`